src/components/encoding-pane/filter-shelf.tsx

```diff
@@ -0,0 +1,125 @@
+import {FieldQuery} from 'compassql/build/src/query/encoding';
+import {Schema} from 'compassql/build/src/schema';
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {ConnectDropTarget, DropTarget, DropTargetCollector, DropTargetSpec} from 'react-dnd';
+import {isOneOfFilter, isRangeFilter, OneOfFilter, RangeFilter} from 'vega-lite/build/src/filter';
+import {FILTER_ADD, FILTER_REMOVE, FilterAction} from '../../actions/filter';
+import {DraggableType} from '../../constants';
+import {ShelfFieldDef} from '../../models/shelf/encoding';
+import {DraggedFieldIdentifier} from '../field';
+import * as styles from './filter-shelf.scss';
+import {OneOfFilterShelf} from './one-of-filter-shelf';
+import {RangeFilterShelf} from './range-filter-shelf';
+
+/**
+ * Props for react-dnd of FilterShelf
+ */
+export interface FilterShelfDropTargetProps {
+  connectDropTarget: ConnectDropTarget;
+
+  isOver: boolean;
+
+  item: Object;
+}
+
+export interface FilterShelfPropsBase {
+  filters: Array<RangeFilter | OneOfFilter>;
```

Comment:
`RangeFilter | OneOfFilter)[]` throws a TSLint warning: [tslint] Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead. (array-type), then it auto-fixes it to Array<RangeFilter | OneOfFilter>